service/history/queues/executable.go

```diff
@@ -277,6 +283,22 @@ func (e *executableImpl) Execute() (retErr error) {
 	)
 	e.Unlock()
 
+	ctx, span := e.tracer.Start(
+		ctx,
+		fmt.Sprintf("queue.Execute/%v", e.GetType().String()),
+		trace.WithSpanKind(trace.SpanKindConsumer),
+		trace.WithAttributes(
+			attribute.Key(telemetry.WorkflowIDKey).String(e.GetWorkflowID()),
+			attribute.Key(telemetry.WorkflowRunIDKey).String(e.GetRunID()),
+			attribute.Key("task-type").String(e.GetType().String()),
+			attribute.Key("task-id").Int64(e.GetTaskID())))
+	defer func() {
+		if retErr != nil {
+			span.RecordError(retErr)
+		}
+		span.End()
+	}()
```

Comment:
This is the main change.

Reply:
I wonder how many extra memory allocation we will have with this even when it's a noopTracer. The allocation for those attributes will happen anyway? e.g. for loggers we have a lazy logger to avoid creating logger tags when it's not needed.
May need to monitor the cpu usage after the change.

Reply:
I was wondering if you would pick up on that, too :) It might be best to wrap the entire thing in an `if` check given how critical this code path is. I'll do that 👍

Reply:
I would add a comment why `if` check is needed.